### ISSUE TITLE
Possible bug: undefined method `resource_collection' for nil:NilClass

### ISSUE THREAD

**Issue by JCotton1123** [chef-cookbooks/compat_resource#116]
### Cookbook version
12.14.7

### Chef-client version
It seems this is broken in chef versions <= 12.13 based on some quick testing. Confirmed its broken in 12.8.1, 12.9.38, and 12.13.37.

### Platform Details
Ubuntu 14.04

### Scenario:
iptables_rule resource fails to be applied but the stacktrace seems to indicate an issue with compat_resource.

### Steps to Reproduce:
Checkout git@github.com:chef-cookbooks/iptables.git, create the below .kitchen.local.yml, and `kitchen converge default-ubuntu-1404`.

```
provisioner:
  name: chef_zero
  require_chef_omnibus: 12.8.1
```

### Expected Result:
No error.

### Actual Result:
```
    ================================================================================
           Error executing action `enable` on resource 'iptables_rule[sshd]'
           ================================================================================

           NoMethodError
           -------------
           undefined method `resource_collection' for nil:NilClass

           Cookbook Trace:
           ---------------
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/dsl/declare_resource.rb:187:in `find_resource!'
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/dsl/declare_resource.rb:218:in `find_resource'
           /tmp/kitchen/cache/cookbooks/iptables/resources/rule.rb:31:in `block (2 levels) in class_from_file'
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/dsl/declare_resource.rb:73:in `with_run_context'
           /tmp/kitchen/cache/cookbooks/iptables/resources/rule.rb:30:in `block in class_from_file'
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:132:in `instance_eval'
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:132:in `compile_and_converge_action'
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'

           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/iptables_test/recipes/default.rb

             3: iptables_rule 'sshd'

           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/iptables_test/recipes/default.rb:3:in `from_file'

           iptables_rule("sshd") do
             action [:enable]
             retries 0
             retry_delay 2
             default_guard_interpreter :default
             declared_type :iptables_rule
             cookbook_name "iptables_test"
             recipe_name "default"
           end
```

**Comment by lamont-granquist** [comment]
sand it looks like that is with the most recent version of the iptables cookbook?  (since i see `with_run_context` in the stacktrace it certainly looks like the code i wrote a couple days ago)

**Comment by lamont-granquist** [comment]
Chef.run_context is nil, i'm not sure where we're patching the Chef class to have the run_context accessor, but I think we're missing the fact that its global state and it needs a patch to load that global state.

**Comment by JCotton1123** [comment]
@lamont-granquist I wasn't sure if this issue was with iptables or compat. Should I move this issue to the iptables repo?

**Comment by lamont-granquist** [comment]
i'm pretty certain its a bug here.

**Comment by lamont-granquist** [comment]
@JCotton1123 does the 12.16.0 release fix the problem?

**Comment by lamont-granquist** [comment]
my testing shows it fixed, i can replicate it on 12.8.1 with the 12.14.7 cookbook version, and if i just change Berksfile.lock to compat_resource 12.16.0 it passes.

**Comment by JCotton1123** [comment]
@lamont-granquist Thanks for fixing this
